internal/model/puller.go

```diff
@@ -341,12 +353,33 @@ func (p *Puller) pullerIteration(ignores *ignore.Matcher) int {
 		return true
 	})
 
+nextFile:
 	for {
 		fileName, ok := p.queue.Pop()
 		if !ok {
 			break
 		}
 		if f, ok := p.model.CurrentGlobalFile(p.folder, fileName); ok {
+			if !f.IsSymlink() {
+				key := string(f.Blocks[0].Hash)
+				for i, candidate := range buckets[key] {
+					if scanner.BlocksEqual(candidate.Blocks, f.Blocks) {
+						// Remove the candidate from the bucket
+						l := len(buckets[key]) - 1
+						buckets[key][i] = buckets[key][l]
+						buckets[key] = buckets[key][:l]
+						// Remove the pending deletion (as we perform it by renaming)
+						delete(fileDeletions, candidate.Name)
+
+						p.renameFile(candidate, f)
+
+						p.queue.Done(fileName)
+						goto nextFile
```

Comment:
I think `continue nextFile` sounds nicer and illustrates that we're taking a new round of a loop, not just going someplace random for vacation. :)